src/storage/blockchain/onchain-issuer.ts

```diff
@@ -0,0 +1,102 @@
+import { DID, Id, chainIDfromDID } from '@iden3/js-iden3-core';
+import { Contract, ethers } from 'ethers';
+import { INonMerklizedIssuerABI as abi } from '@iden3/onchain-non-merklized-issuer-base-abi';
+import { Options } from '@iden3/js-jsonld-merklization';
+import { W3CCredential } from '../../verifiable';
+import { OnchainNonMerklizedIssuerAdapter } from './onchain-issuer-adapter/non-merklized/version/v0.0.1/onchain-non-merklized-issuer-adapter';
+import { EthConnectionConfig } from './state';
+import { IOnchainIssuer } from '../interfaces/onchain-issuer';
+
+/**
+ * Represents an adapter for interacting with on-chain issuers.
+ *
+ * @public
+ * @beta
+ * @class OnchainIssuer
+ */
+export class OnchainIssuer implements IOnchainIssuer {
+  private readonly _ethConnectionConfig: EthConnectionConfig[];
+  private readonly _merklizationOptions?: Options;
+
+  /**
+   * Initializes an instance of `Adapter`.
+   * @param config The configuration for the Ethereum connection.
+   * @param merklizationOptions Optional settings for merklization.
+   */
+  constructor(config: EthConnectionConfig[], options?: Options) {
```

Comment:
do the same as you did for adapter, options is an object of separate type (e.g. OnchainIssuerOptions and merklizationOptions is a property there)